addons/csw/functions/fnc_getAvailableAmmo.sqf

```diff
@@ -0,0 +1,62 @@
+#include "..\script_component.hpp"
+/*
+ * Author: LinkIsGrim
+ * Gets available ammo for a CSW.
+ *
+ * Arguments:
+ * 0: CSW <OBJECT>
+ * 1: Only loaded magazines <BOOL> (default: false)
+ * 2: Skip ammo from vehicles <BOOL> (default: true)
+ * 3: Include CSW crew <BOOL> (default: true)
+ *
+ * Return Value:
+ * Available Ammo <HASHMAP>
+ *   Magazine classname <STRING>
+ *   Total Ammo <NUMBER>
+ *
+ * Example:
+ * [cursorObject] call ace_csw_fnc_getAvailableAmmo
+ *
+ * Public: Yes
+ */
+
+params [["_vehicle", objNull, [objNull]], ["_onlyLoaded", false, [false]], ["_skipVehicles", true, [true]], ["_includeCrew", true, [true]]];
+
+if (isNull _vehicle) exitWith {createHashMap};
+
+private _availableMagazines = createHashMap;
+
+private _fnc_addAmmo = {
+    params ["_magazine", "_ammo"];
+    if !(_magazine in _availableMagazines) then {
+        _availableMagazines set [_magazine, _ammo];
+    } else {
+        _availableMagazines set [_magazine, (_availableMagazines get _magazine) + _ammo];
+    };
```

Comment:
I haven't tested the code for speed, but I imagine the suggested code is faster:
```suggestion
    _availableMagazines set [_magazine, (_availableMagazines getOrDefault [_magazine, 0]) + _ammo];
```
At this point I'd remove the function altogether and place the one line of code directly where necessary, even if it's duplicate.